packetbeat/protos/mysql/mysql.go

```diff
@@ -160,6 +175,13 @@ func (mysql *mysqlPlugin) init(results protos.Reporter, config *mysqlConfig) err
 		mysql.transactionTimeout,
 		protos.DefaultTransactionHashSize)
 	mysql.transactions.StartJanitor(mysql.transactionTimeout)
+
+	// preparement cache
+	mysql.preparestatements = common.NewCache(
+		mysql.preparestatementTimeout,
+		protos.DefaultTransactionHashSize)
+	mysql.preparestatements.StartJanitor(mysql.preparestatementTimeout)
```

Comment:
Do we really want to timeout prepared statements? Maybe have timeout only active on prepared statements after packet loss only (and only for N seconds). If a prepared statement is lost in the cache, we can not emit the original query in the event.

MySql/MariaDB itself enforces a limit on total statements one can use. So application using large number of prepared statements either create/close statements quite often or use an internal cache of active prepared statements.

Reply:
I create a statement map for each flow in CACHE.
All the prepare statements of a flow will be cached in a  same map.
So there will be more and more maps in CACHE. I can delete the map when receive the Fin packet of a flow, but sometimes Fin packet lost or not be sniffed.
So I add the timeout method to control the size of the CACHE.
The cache expiration time will be overwritten after each put or get method.

Reply:
makes sense.